### ISSUE TITLE
Currently unsupported features and caveats (aka the big TODO list)

### ISSUE THREAD

**Issue by EiffL** [DifferentiableUniverseInitiative/jaxDecomp#1]
Because implementing all features is not necessarily interesting unless there is a need for it, here are the current restrictions of the code in its current version. **All these restrictions can be lifted if you have a need for them**, don't hesitate to comment on this issue if there is something you would like to be able to do!

**General**
- [x]  Configuration mechanism to choose the communication API (by default CUDA aware MPI) only works at initialization
- [x] No interface to run autotuning to figure out the best communication strategy
- [x] No interface with the JAX 0.4 Array API

**Transpose operations**
- [x] Only transpose operations that preserve the size of local slices are supported
- [x] Transpose ops do not have batching or gradient operations implemented
- [x] No support for letting XLA allocate the workspace size needed for the transpose
- [x] Double precision is silently not supported (returns crazy values!)

**FFTs**
- [x] Only complex FFTs are implemented as a single CUDA-level operation
- [x] Only FFTs operations  that preserve the size of local slices are supported

**Comment by ASKabalan** [comment]
Adding to the todo list 

- [ ]  update CMake for cuda 12 ; starting jax [0.4.26](https://jax.readthedocs.io/en/latest/changelog.html#jaxlib-0-4-26) cuda11 support is dropped

**Comment by ASKabalan** [comment]
Everything in here is implemented 
Follow-up todo list is https://github.com/DifferentiableUniverseInitiative/jaxDecomp/issues/37
